src/main/java/felix/utils/Parser.java

```diff
@@ -0,0 +1,101 @@
+package felix.utils;
+
+import java.time.format.DateTimeParseException;
+
+import felix.command.AddTaskCommand;
+import felix.command.Command;
+import felix.command.DeleteCommand;
+import felix.command.ExitCommand;
+import felix.command.FindCommand;
+import felix.command.ListCommand;
+import felix.command.MarkCommand;
+import felix.command.UnmarkCommand;
+import felix.exception.FelixException;
+import felix.task.Deadline;
+import felix.task.Event;
+import felix.task.Task;
+import felix.task.ToDo;
+
+/**
+ * Class representing a parser that processes user commands
+ */
+public class Parser {
+    /**
+     * Parses a line of user input and returns the corresponding Command.
+     *
+     * @return Command.
+     * @throws FelixException If command is unrecognised or command formatting is wrong.
+     */
+    public Command getCommand(String line) throws FelixException {
```

Comment:
Do you think its better to split this long method (more than 30 lines) into smaller more readable methods?